### PR TITLE
Fix precision error

### PR DESCRIPTION
**Pull request by JaGeo**
## Summary
This is an WIP attempt to fix the precision error mentioned in https://github.com/materialsproject/atomate2/issues/166 (@utf )

It currently has tons of test failures as one can imagine. I think we also need to recreate a huge part of the test data. I am not sure I have even all programs required to do so. I would be very happy about help in this regard. All VASP flows fail and further tests as I added the contcar to some methods/functions.

### CODE REVIEW

File: src/atomate2/vasp/schemas/calculation.py [hunk 1]
```diff
@@ -493,7 +497,8 @@ def from_vasp_outputs(
         outcar_dict = outcar.as_dict()
         outcar_dict.pop("run_stats")
 
-        structure = vasprun.final_structure
+        # structure = vasprun.final_structure
```
Comment (utf) [inline src/atomate2/vasp/schemas/calculation.py:500]: Can you change this line to:
```suggestion
        # use structure from CONTCAR as it is written to greater precision than in the vasprun
```

**Review by utf — COMMENTED** [review]
Thanks! 

A couple of minor comments now. Will take another look after the tests run.

File: src/atomate2/vasp/schemas/task.py [hunk 2]
```diff
@@ -649,8 +650,11 @@ def _get_task_files(files, suffix=""):
         for file in files:
             if file.match(f"*vasprun.xml{suffix}*"):
                 vasp_files["vasprun_file"] = file
+                vasp_files["vasprun_file"] = file
```
[the remark this answers is not in the document]
Comment (JaGeo) [reply src/atomate2/vasp/schemas/task.py:653]: No, you are correct, of course. Will remove it.

**Comment by JaGeo** [comment]
I have added the suggestions. As this is quite a drastic change, I am happy about any kind of suggestion. I might also be overlooking something with regard to CONTCAR fetching ...

**Comment by utf** [comment]
So hopefully we won't have to re-generate the test data. It looks at first glance that the fixes should be relatively simple.

Most of the errors look like an issue with sorting the contcars?

```
else sorted(contcars)[-1]
IndexError: list index out of range
```
Does your code definitely work with gzipped contacts for example?

In the schema tests (`atomate2/tests/vasp/schemas`) there are many:
```
        files = test_object.task_files[task_name]
>       calc_doc, _ = Calculation.from_vasp_files(dir_name, task_name, **files)
E       TypeError: from_vasp_files() missing 1 required positional argument: 'contcar_file'
```

These can be easily addressed by adding the path to the contcar file to the test dicts, e.g., [here](https://github.com/materialsproject/atomate2/blob/4dbeccb98f12968ba294cb34af345f03c7a497a1/tests/vasp/schemas/conftest.py#L49) and any other `task_files` dicts in that file.

Similarly, there are errors such as:
```
        test_object = get_test_object(object_name)
        folder = vasp_test_dir / test_object.folder / "outputs"
        vasprun_file = folder / test_object.task_files[task_name]["vasprun_file"]
        outcar_file = folder / test_object.task_files[task_name]["outcar_file"]
        vasprun = Vasprun(vasprun_file)
        outcar = Outcar(outcar_file)
>       test_doc = CalculationOutput.from_vasp_outputs(vasprun, outcar)
E       TypeError: from_vasp_outputs() missing 1 required positional argument: 'contcar'
```
These can be fixed by adding code such as:
```
        contcar_file = folder / test_object.task_files[task_name]["contcar_file"]
        contcar = Poscar.from_file(contcar_file)
        test_doc = CalculationOutput.from_vasp_outputs(vasprun, outcar, contcar)
```

**Comment by JaGeo** [comment]
I don't think it has anything to do with the gzipping. I immitated the OUTCAR and vasprun.xml reading exactly. 
Files are copied before via vasp.files and it wasn't copying the CONTCAR. 
Now, the existing CONTCAR however leads to different errors. I will further look into it and hopefully have it resolved by the end of the day.

**Comment by JaGeo** [comment]
Found the issue. The CONTCAR is already a renamed POSCAR here. This is then used to read the structure ... I think I fixed some more issues ...

**Comment by JaGeo** [comment]
Once all tests are passing, I will do a double relaxation to check that POSCAR and CONTCAR copying is not somehow messed up now. Would be happy if you could have a look as well as I am worried to have overlooked something.

**Comment by JaGeo** [comment]
Should we add an additional tag to check if CONTCAR has been copied to POSCAR before reading? I am a bit unsure about the best approach here as I feel this could lead to a lot of problems as copy_vasp_outputs has the option to not copy contcar to poscar.

**Comment by utf** [comment]
Thinking about this a bit more, the structure the each calculation uses never actually comes from the POSCAR/CONTCAR in the current directory. Instead it comes from the structure that gets passed in to the make function. E.g., it comes from the structure that is stored in the output of the previous job.

The copying CONTCAR from previous directory stuff is hangover from atomate1 where the structure was passed by copying files. One option is remove the copying, although it could potentially be useful when we implement checkpointing functionality (see #156).

Maybe for now we can leave it in there. Either way, that bit of the code will not impact which structure gets used.

**Comment by JaGeo** [comment]
Okay, great. 
I will be rerunning my phonon workflow for Mg3Sb2 now and check if all BORN charge symmetry problems have now vanished. I hope they have ...

**Comment by utf** [comment]
Thanks for being so thorough! Hope it fixes this issue too.

**Comment by JaGeo** [comment]
Born charges are computed correctly but there is still another issue in the phonon workflow for this structure... Will continue checking. ..

**Comment by utf** [comment]
Does this fix the Born problem? Am I ok to merge?

**Comment by JaGeo** [comment]
I think it fixes the general issue that the precision of the CONTCAR files gets lost within a flow which could influence the results. Another issue is now actually showing up in the phonon workflow that wasn't there before. I am currently fixing it.... Sorry for this. These problems did not show up in the very symmetric structures ...

File: tests/vasp/conftest.py [hunk 3]
```diff
@@ -256,6 +257,7 @@ def copy_vasp_outputs(ref_path: Union[str, Path]):
     import shutil
 
     output_path = ref_path / "outputs"
+    print(output_path.iterdir())
```
Comment (utf) [inline tests/vasp/conftest.py:260]: Please remove

**Review by utf — COMMENTED** [review]
Its great that you're being so thorough with testing the workflow. I'm very grateful that you've caught these bugs now rather than further down the line. 

If you don't mind, I will merge this PR now (once that stray debugging line is removed), as to avoid complicating the PR.

File: tests/vasp/conftest.py (hunk 3 above)
Comment (JaGeo) [reply tests/vasp/conftest.py:260]: Will do after a meeting :)

**Comment by utf** [comment]
Thank you :)

**Pull request merged**